libfwupdplugin/fu-common-version.c

```diff
@@ -449,6 +449,29 @@ fu_common_version_verify_format (const gchar *version,
 	return TRUE;
 }
 
+/**
+ * fu_common_vercmp_full:
+ * @version_a: the semver release version, e.g. 1.2.3
+ * @version_b: the semver release version, e.g. 1.2.3.1
+ * @fmt: a #FwupdVersionFormat, e.g. %FWUPD_VERSION_FORMAT_PLAIN
+ *
+ * Compares version numbers for sorting taking into account the version format
+ * if required.
+ *
+ * Returns: -1 if a < b, +1 if a > b, 0 if they are equal, and %G_MAXINT on error
+ *
+ * Since: 1.3.9
+ */
+gint
+fu_common_vercmp_full (const gchar *version_a,
+		       const gchar *version_b,
+		       FwupdVersionFormat fmt)
```

Comment:
Another thought here; since you have the version format information, would it make sense to count the dots to validate a and b are (conceivably) the same format?

Reply:
I also wondered this; but would testing a kernel version of `2.6.1.1 > 2.6.1` be then invalid?

Reply:
Do distros add another digit there for their builds?  On Ubuntu my `uname -r` has the extra number as a dash (eg 5.3.0-18)

Reply:
Fair point, mine is a triplet too.

Reply:
Lets add this kind of checking for now; I don't want to introduce a regression. There's also no `GError` (or any way to return an error!) so I'm no sure how we'd signify an "invalid triplet" for instance.